packages/test-runner-cli/src/reporter/junitReporter.ts

```diff
@@ -0,0 +1,164 @@
+import path from 'path';
+import fs from 'fs';
+import { Reporter, ReporterArgs, TestResult, TestRunnerCoreConfig, TestSession } from '@web/test-runner-core';
+
+import { createSourceMapFunction, SourceMapFunction } from '../utils/createSourceMapFunction';
+import { createStackLocationRegExp } from '../utils/createStackLocationRegExp';
+
+import XML from 'xml';
+
+export interface JUnitReporterArgs {
+  outputPath?: string;
+}
+
+type TestSessionMetadata =
+  Omit<TestSession, 'tests'>;
+
+type TestResultWithMetadata =
+  TestResult & TestSessionMetadata;
+
+type TestResultsWithMetadataByBrowserName = {
+  [key: string]: TestResultWithMetadata[];
```

Comment:
I prefer `Record<string, TestResultWithMetadata[]>`, what do you think?

Reply:
done